src/main/java/com/avispl/dal/communicator/polycom/videoos/PolycomVideoOS.java

```diff
@@ -688,7 +783,23 @@ private CallStatus generateCallStatus(String callId, CallStatus.CallStatusState
     }
 
     private RegistrationStatus retrieveRegistrationStatus() throws Exception {
-        RegistrationStatus registrationStatus = new RegistrationStatus();
+        RegistrationStatus registrationStatus = null;
+        Boolean deviceMode = retrieveDeviceMode();
+        if (deviceMode != null && deviceMode) {
+            if (localEndpointStatistics != null) {
+                registrationStatus = localEndpointStatistics.getRegistrationStatus();
+            }
+            if (logger.isDebugEnabled()) {
+                logger.debug("The codec is in the device mode. Skipping registration status retrieval, falling back to the cached values");
+            }
+            if (registrationStatus == null) {
+                registrationStatus = new RegistrationStatus();
+            }
+            registrationStatus.setSipRegistered(true);
+            registrationStatus.setH323Registered(true);

```

Comment:
So, if codec is in `deviceMode` SIP and H323 registration is always true?

Reply:
correct. Basically, these statuses do not make sense for this codec when it's in the device mode, but to avoid producing false-positive error reports whenever the device enters the device mode, it is set to "true", so if there are any registration errors present in a codec mode, they would resolve for the time the codec is in the device mode.